tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet.md

```diff
@@ -0,0 +1,292 @@
+
+- Rust installation - the node template is written in [Rust](https://www.rust-lang.org/){target=\_blank}. Install it by following the [Installation](/develop/parachains/get-started/install-polkadot-sdk){target=\_blank} guide for step-by-step instructions on setting up your development environment
+
+## Download and Compile the Node Template
+
+The [Polkadot SDK Solochain Template](https://github.com/paritytech/polkadot-sdk-solochain-template){target=\_blank} provides a ready-to-use development environment for building using the [Polkadot SDK](https://github.com/paritytech/polkadot-sdk){target=\_blank}. Follow these steps to compile the node:
+
+1. Clone the repository:
+    ```bash
+    git clone -b {{dependencies.polkadot_sdk_solochain_template.version}} {{dependencies.polkadot_sdk_solochain_template.repository_url}}
+    ```
+
+    !!!note
+        Ensure you're using the version `{{dependencies.polkadot_sdk_solochain_template.version}}` of the Polkadot SDK Solochain Template to be able to follow this tutorial step by step.
+
+2. Navigate to the project directory:
+    ```bash
+    cd polkadot-sdk-solochain-template
+    ```
+
+3. Compile the node template:
+    ```bash
+    cargo build --release
+    ```
+
+    !!!note
+        Depending on your machine's specifications, initial compilation may take several minutes. For optimized artifacts, use the `--release` flag.
+
+4. Verify successful compilation by checking that the output is similar to:
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/compilation-output.html'
+
+## Create a New Project
+
+In this tutorial, you'll build a custom pallet from scratch to demonstrate the complete workflow, rather than starting with the pre-built pallet-template. The first step is to create a new Rust package for your pallet:
+
+1. Navigate to the `pallets` directory in your workspace:
+
+    ```bash
+    cd pallets
+    ```
+
+2. Create a new Rust library project for your custom pallet by running the following command:
+
+    ```bash
+    cargo new --lib custom-pallet
+    ```
+
+3. Enter the new project directory:
+
+    ```bash
+    cd custom-pallet
+    ```
+
+4. Ensure the project was created successfully by checking its structure. The file layout should resemble the following:
+
+    ```
+    custom-pallet 
+    ├── Cargo.toml
+    └── src
+        └── lib.rs
+    ```
+
+    If the files are in place, your project setup is complete, and you're ready to start building your custom pallet.
+
+## Add Dependencies
+
+To build and integrate your custom pallet into a Polkadot SDK-based runtime, you must add specific dependencies to the `Cargo.toml` file.  These dependencies provide essential modules and features required for pallet development. Since your custom pallet is part of a workspace that includes other components, such as the runtime, the configuration must align with the workspace structure. Follow the steps below to set up your `Cargo.toml` file properly:
```

Comment:
As this is the first time you have talked directly about the Cargo. some of the pallet, I think you can say `custom-pallet/Cargo.toml` and then refer to it just as it is now. Just for the sake of avoiding any misunderstood with the runtime's Cargo.toml